### PR TITLE
Add BackupMetadata struct, move ZMVersion from SyncEngine

### PR DESCRIPTION
**Pull request by daehn**
## What's new in this PR?

* Add BackupMetadata struct + validation methods.
* Move ZMVersion from `wire-ios-sync-engine` + convert it to Swift in the process.

### CODE REVIEW

File: Source/ManagedObjectContext/BackupMetadata.swift [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+public extension BackupMetadata {
+    
+    enum VerificationError: Error {
+        case backupFromNewerAppVersion
+        case noVersion
+        case clientMismatch
+        case userMismatch
+    }
+    
+    func verify(using client: UserClient) throws {
```
Comment (vytis) [inline Source/ManagedObjectContext/BackupMetadata.swift]: I think we will need to create a new client on import anyway, so this is probably not needed.  Also this method does nothing except throwing, maybe it's then better to just return optional error?

File: Source/Utilis/ZMVersion.m [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+
+#import "ZMVersion.h"
+
+@import Foundation;
+@import WireUtilities;
+
+@interface ZMVersion ()
```
Comment (vytis) [inline Source/Utilis/ZMVersion.m]: Should probably move tests as well

File: Source/ManagedObjectContext/BackupMetadata.swift (hunk 1 above)
Comment (daehn) [reply Source/ManagedObjectContext/BackupMetadata.swift]: Good point!

File: Source/Utilis/ZMVersion.m (hunk 2 above)
Comment (daehn) [reply Source/Utilis/ZMVersion.m]: Done in d34f5da ;)

File: Source/ManagedObjectContext/BackupMetadata.swift (hunk 1 above)
Comment (daehn) [reply Source/ManagedObjectContext/BackupMetadata.swift]: @vytis updated!

**Comment by daehn** [comment]
Tests on SE are failing with these changes, I need to investigate.

**Pull request merged**